runhouse/resources/hardware/cluster.py

```diff
@@ -432,21 +432,30 @@ def _sync_runhouse_to_cluster(self, _install_url=None, env=None):
                 filter_options="- docs/",
             )
             rh_install_cmd = "python3 -m pip install ./runhouse"
+
+            # installing skypilot on cluster for .keep_warm of on-demand cluster.
+            sky_install_cmd = "python3 -m pip install skypilot"
```

Comment:
why do we need to install sky here? this is used across various cluster types, i don't think we should enforce needing to have it installed?